File: Extensions/Android_Gradle/app/src/main/java/Extensions/CExtLoad.java
/* Copyright (c) 1996-2013 Clickteam
 *
 * This source code is part of the Android exporter for Clickteam Multimedia Fusion 2.
 * 
 * Permission is hereby granted to any person obtaining a legal copy 
 * of Clickteam Multimedia Fusion 2 to use or modify this source code for 
 * debugging, optimizing, or customizing applications created with 
 * Clickteam Multimedia Fusion 2.  Any other use of this source code is prohibited.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
 * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
 * IN THE SOFTWARE.
 */
//----------------------------------------------------------------------------------
//
// CEXTLOADER: Chargement des extensions
//
//----------------------------------------------------------------------------------
package Extensions;

import Runtime.Log;
import Runtime.MMFRuntime;
import Services.CFile;

public class CExtLoad
{
	short handle;
	String name;
	String subType;

	public CExtLoad()
	{
	}

	public void loadInfo(CFile file)
	{
		long debut = file.getFilePointer();

		short size = (short) Math.abs (file.readAShort());
		handle = file.readAShort();
		file.skipBytes(12);
		name = file.readAString();
		int pos = name.lastIndexOf('.');
		name = name.substring(0, pos);
		int index = name.indexOf('-');
		while (index > 0)
		{
			name = name.substring(0, index) + '_' + name.substring(index + 1);
			index = name.indexOf('-');
		}
		index = name.indexOf(' ');
		while (index > 0)
		{
			name = name.substring(0, index) + '_' + name.substring(index + 1);
			index = name.indexOf(' ');
		}
		subType = file.readAString();

		file.seek(debut + size);
	}

	public CRunExtension loadRunObject()
	{
		CRunExtension object=null;

		if (MMFRuntime.nativeExtensions.contains ("CRun" + name))
		{
			object = new CRunNativeExtension ("CRun" + name);
		}
		else if (name.compareToIgnoreCase("Android")==0)
		{
			object=new CRunAndroid();
		}
		else if (name.compareToIgnoreCase("OUYA")==0)
		{
			object=new CRunOUYA();
		}
		else
		{
			// STARTCUT
			// STARTCUT
			// STARTCUT
			if (name.compareToIgnoreCase("TapTap")==0)
			{
				object=new CRunTapTap();
			}


//			if (name.compareToIgnoreCase("clickteam_movement_controller")==0)
//			{
//				object=new CRunclickteam_movement_controller();
//			}
//			if (name.compareToIgnoreCase("kcarray")==0)
//			{
//				object=new CRunKcArray();
//			}
//			if (name.compareToIgnoreCase("Layer")==0)
//			{
//				object=new CRunLayer();
//			}
//
//			if (name.compareToIgnoreCase("KcBoxA")==0)
//			{
//				object=new CRunKcBoxA();
//			}
//			if (name.compareToIgnoreCase("KcBoxB")==0)
//			{
//				object=new CRunKcBoxB();
//			}
//			if (name.compareToIgnoreCase("parser")==0)
//			{
//				object=new CRunparser();
//			}
//			if (name.compareToIgnoreCase("Accelerometer")==0)
//			{
//				object=new CRunAccelerometer();
//			}
//			if (name.compareToIgnoreCase("kcbutton")==0)
//			{
//				object=new CRunKcButton();
//			}
//			if (name.compareToIgnoreCase("KcEdit")==0)
//			{
//				object=new CRunkcedit();
//			}
//			if (name.compareToIgnoreCase("KcList")==0)
//			{
//				object=new CRunkclist();
//			}
//			if (name.compareToIgnoreCase("KcCombo")==0)
//			{
//				object=new CRunkccombo();
//			}
//			if (name.compareToIgnoreCase("kchisc")==0)
//			{
//				object=new CRunkchisc();
//			}
//			if (name.compareToIgnoreCase("kcfile")==0)
//			{
//				object=new CRunkcfile();
//			}
//			if (name.compareToIgnoreCase("KcClock")==0)
//			{
//				object=new CRunkcclock();
//			}
			if (name.compareToIgnoreCase("kcpica")==0)
			{
				object=new CRunkcpica();
			}
//			if (name.compareToIgnoreCase("AdvDir")==0)
//			{
//				object=new CRunAdvDir();
//			}
//			if (name.compareToIgnoreCase("FontPacker")==0)
//			{
//				object=new CRunFontPacker();
//			}
//			if (name.compareToIgnoreCase("ACamera")==0)
//			{
//				object=new CRunACamera();
//			}
//			if (name.compareToIgnoreCase("DlgMsgBox")==0)
//			{
//				object=new CRunDlgMsgBox();
//			}
//			//if (name.compareToIgnoreCase("AndroidEmail")==0)
//			//{
//			//	object=new CRunAndroidEmail();
//			//}
//			//if (name.compareToIgnoreCase("EmailSender")==0)
//			//{
//			//	object=new CRunEmailSender();
//			//}
//			//if (name.compareToIgnoreCase("OneSignalApp")==0)
//			//{
//			//	object=new CRunOneSignalApp();
//			//}
//
//			if (name.compareToIgnoreCase("MultipleTouch")==0)
//			{
//				object=new CRunMultipleTouch();
//			}
//			if (name.compareToIgnoreCase("JoystickControl")==0)
//			{
//				object=new CRunJoystickControl();
//			}
//			if (name.compareToIgnoreCase("ActiveBackdrop")==0)
//			{
//				object=new CRunActiveBackdrop();
//			}
//			if (name.compareToIgnoreCase("curl")==0)
//			{
//				object=new CRunCurl();
//			}
			if (name.compareToIgnoreCase("kcini")==0)
			{
				object=new CRunkcini();
			}
//			if (name.compareToIgnoreCase("kcwctrl")==0)
//			{
//				object=new CRunkcwctrl();
//			}
//			if (name.compareToIgnoreCase("CalcRect")==0)
//			{
//				object=new CRunCalcRect();
//			}
//			if (name.compareToIgnoreCase("Easing")==0)
//			{
//				object=new CRunEasing();
//			}
//			if (name.compareToIgnoreCase("StringTokenizer")==0)
//			{
//				object=new CRunStringTokenizer();
//			}
//			if (name.compareToIgnoreCase("Platform")==0)
//			{
//				object=new CRunPlatform();
//			}
//			if (name.compareToIgnoreCase("WebView2")==0)
//			{
//				object=new CRunWebView2();
//			}
//			if (name.compareToIgnoreCase("AndroidUpdater")==0)
//			{
//				object=new CRunAndroidUpdater();
//			}
//			if (name.compareToIgnoreCase("AndroidSplash")==0)
//			{
//				object=new CRunAndroidSplash();
//			}
//			if (name.compareToIgnoreCase("Oldini")==0)
//			{
//				object=new CRunOldini();
//			}
//			if (name.compareToIgnoreCase("GMaps")==0)
//			{
//				object=new CRunGMaps();
//			}
//			if (name.compareToIgnoreCase("AdMob")==0)
//			{
//				object=new CRunAdMob();
//			}
//			if (name.compareToIgnoreCase("LeadBolt")==0)
//			{
//				object=new CRunLeadBolt();
//			}
//			if (name.compareToIgnoreCase("Location")==0)
//			{
//				object=new CRunLocation();
//			}
			if (name.compareToIgnoreCase("Get")==0)
			{
				object=new CRunGet();
			}
//			if (name.compareToIgnoreCase("InputType")==0)
//			{
//				object=new CRunInputType();
//			}
//			if (name.compareToIgnoreCase("MTRandom")==0)
//			{
//				object=new CRunMTRandom();
//			}
//			if (name.compareToIgnoreCase("AndroidPlus")==0)
//			{
//				object=new CRunAndroidPlus();
//			}
//			if (name.compareToIgnoreCase("AndroidDialog")==0)
//			{
//				object=new CRunAndroidDialog();
//			}
//			if (name.compareToIgnoreCase("InAppAndroid")==0)
//			{
//				object=new CRunInAppAndroid();
//			}
//			if (name.compareToIgnoreCase("Inventory")==0)
//			{
//				object=new CRunInventory();
//			}
//			if (name.compareToIgnoreCase("FIRETV")==0)
//			{
//				object=new CRunFIRETV();
//			}
//
//			if (name.compareToIgnoreCase("NEXUSTV")==0)
//			{
//				object=new CRunNEXUSTV();
//			}
//			if (name.compareToIgnoreCase("JSON_Object")==0)
//			{
//				object=new CRunJSON_Object();
//			}
//
//			if (name.compareToIgnoreCase("GetLocale")==0)
//			{
//				object=new CRunGetLocale();
//			}
//
//			if (name.compareToIgnoreCase("Lacewing")==0)
//			{
//				object=new CRunLacewing();
//			}
//
//			if (name.compareToIgnoreCase("VideoAndroid")==0)
//			{
//				object=new CRunVideoAndroid();
//			}
//
//			//if (name.compareToIgnoreCase("AVideoSurface")==0)
//			//{
//			//	object=new CRunAVideoSurface();
//			//}
//
//			if (name.compareToIgnoreCase("CreateByName")==0)
//			{
//				object=new CRunCreateByName();
//			}
//
//			if (name.compareToIgnoreCase("ScreenZoom")==0)
//			{
//				object=new CRunScreenZoom();
//			}
//
//			if (name.compareToIgnoreCase("ChartBoost")==0)
//			{
//				object=new CRunChartBoost();
//			}
//
//			if (name.compareToIgnoreCase("EncoderX")==0)
//			{
//				object=new CRunEncoderX();
//			}
//
//			if (name.compareToIgnoreCase("Scan")==0)
//			{
//				object=new CRunScan();
//			}
//
//			if (name.compareToIgnoreCase("GPGConnect")==0)
//			{
//				object=new CRunGPGConnect();
//			}
//
//			if (name.compareToIgnoreCase("Perspective")==0)
//			{
//				object=new CRunPerspective();
//			}
//
//			if (name.compareToIgnoreCase("GPGCloud")==0)
//			{
//				object=new CRunGPGCloud();
//			}
//
//
//			if (name.compareToIgnoreCase("GPGTurnbase")==0)
//			{
//				object=new CRunGPGTurnbase();
//			}
//
//
//			if (name.compareToIgnoreCase("GPGAchievement")==0)
//			{
//				object=new CRunGPGAchievement();
//			}
//
//
//			if (name.compareToIgnoreCase("GPGLeaderboard")==0)
//			{
//				object=new CRunGPGLeaderboard();
//			}
//
//
//			if (name.compareToIgnoreCase("GPGPlayers")==0)
//			{
//				object=new CRunGPGPlayers();
//			}
//
//			if (name.compareToIgnoreCase("GPGQuests")==0)
//			{
//				object=new CRunGPGQuests();
//			}
//
//			if (name.compareToIgnoreCase("Box2DBase")==0)
//			{
//				object=new CRunBox2DBase();
//			}
//			if (name.compareToIgnoreCase("Box2DFan")==0)
//			{
//				object=new CRunBox2DFan();
//			}
//			if (name.compareToIgnoreCase("Box2DMagnet")==0)
//			{
//				object=new CRunBox2DMagnet();
//			}
//			if (name.compareToIgnoreCase("Box2DTreadmill")==0)
//			{
//				object=new CRunBox2DTreadmill();
//			}
//			if (name.compareToIgnoreCase("Box2DParticules")==0)
//			{
//				object=new CRunBox2DParticules();
//			}
//			if (name.compareToIgnoreCase("Box2DRopeAndChain")==0)
//			{
//				object=new CRunBox2DRopeAndChain();
//			}
//            if (name.compareToIgnoreCase("Box2DGround")==0)
//            {
//                object=new CRunBox2DGround();
//            }
//            if (name.compareToIgnoreCase("Box2DJoint")==0)
//            {
//                object=new CRunBox2DJoint();
//            }
			// ENDCUT
		}

		if (object!=null)
		{
			Log.Log("Created extension: " + name);
			return object;
		}

		Log.Log("*** MISSING EXTENSION: " + name);
		return null;
	}
}

File: Extensions/iOS/Classes/Extensions/CExtLoad.m
/* Copyright (c) 1996-2014 Clickteam
*
* This source code is part of the iOS exporter for Clickteam Multimedia Fusion 2
* and Clickteam Fusion 2.5.
* 
* Permission is hereby granted to any person obtaining a legal copy 
* of Clickteam Multimedia Fusion 2 or Clickteam Fusion 2.5 to use or modify this source 
* code for debugging, optimizing, or customizing applications created with 
* Clickteam Multimedia Fusion 2 and/or Clickteam Fusion 2.5. 
* Any other use of this source code is prohibited.
*
* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
* FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
* IN THE SOFTWARE.
*/
//----------------------------------------------------------------------------------
//
// CEXTLOAD Chargement des extensions
//
//----------------------------------------------------------------------------------
#import "CExtLoad.h"
#import "CFile.h"
#import "CRunExtension.h"

//F01
#import "CRunGDPRConsent.h"
#import "CRunKcArray.h"
#import "CRunActiveBackdrop.h"
#import "CRunAdvGameBoard.h"
#import "CRunAdvPathMov.h"
#import "CRunCalcRect.h"
#import "CRunIIF.h"
#import "CRunInAndOutController.h"
#import "CRunkcdirect.h"
#import "CRunLayer.h"
#import "CRunMoveSafely2.h"
#import "CRunPlatform.h"
#import "CRunclickteam_movement_controller.h"
#import "CRunKcBoxA.h"
#import "CRunKcBoxB.h"
#import "CRunAccelerometer.h"
#import "CRunLocation.h"
#import "CRunkcini.h"
#import "CRunMultipleTouch.h"
#import "CRuniOSButton.h"
#import "CRuniOSSingleEdit.h"
#import "CRuniOSMultipleEdit.h"
#import "CRunJoystickControl.h"
#import "CRunAdvDir.h"
#import "CRunkcclock.h"
#import "CRunkchisc.h"
#import "CRunkcrandom.h"
#import "CRunObjectMover.h"
#import "CRunWargameMap.h"
#import "CRunparser.h"
#import "CRunMTRandom.h"
#import "CRunGet.h"
#import "CRunkclist.h"
#import "CRunkcinput.h"
#import "CRuniOSStore.h"
#import "CRunStringTokenizer.h"
#import "CRunMoveIt.h"
#import "CRunEasing.h"
#import "CRunkcedit.h"
#import "CRuniOS.h"
#import "CRunkcbutton.h"
#import "CRunCamera.h"
#import "CRunGameCenterConnect.h"
#import "CRunGameCenterLeaderboard.h"
#import "CRunGameCenterAchievements.h"
#import "CRunGameCenterAchievement.h"
#import "CRunGameCenterMultiplayer.h"
#import "CRunkcwctrl.h"
#import "CRunkcpica.h"
#import "CRuniOSVideo.h"
#import "CRunForEach.h"
#import "CRunCreateByName.h"
#import "CRunInventory.h"
#import "CRunDeadReckoning.h"
#import "CRunFlocking.h"
#import "CRunScreenZoom.h"
#import "CRunBox2DBase.h"
#import "CRunBox2DFan.h"
#import "CRunBox2DMagnet.h"
#import "CRunBox2DTreadmill.h"
#import "CRunBox2DParticules.h"
#import "CRunBox2DRopeAndChain.h"
#import "CRunBox2DGround.h"
#import "CRunBox2DJoint.h"
#import "CRunWebView2.h"
#import "CRunStencilMask.h"

//F01END

@implementation CExtLoad

-(void)dealloc
{
	[name release];
	[subType release];
	[super dealloc];
}
-(void)loadInfo:(CFile*)file
{
	NSUInteger debut = [file getFilePointer];
	
	short size = abs([file readAShort]);
	handle = [file readAShort];
	[file skipBytes:12];
	
	name = [file readAString];
	NSRange index = [name rangeOfString:@"."];
	if (index.location!=NSNotFound)
	{
		NSString* temp=[name substringToIndex:index.location];
		[name release];
		name=temp;
		[name retain];
	}
	subType = [file readAString];
	
	[file seek:debut + size];
}

-(CRunExtension*)loadRunObject 
{
	CRunExtension* object=nil;
	
//F02 			
	
if ([name caseInsensitiveCompare:@"GDPRConsent"]==0)
{
object=[[CRunGDPRConsent alloc] init];
}
if ([name caseInsensitiveCompare:@"kcarray"]==0)
	{
		object=[[CRunKcArray alloc] init];
	}
	if ([name caseInsensitiveCompare:@"ActiveBackdrop"]==0)
	{
		object=[[CRunActiveBackdrop alloc] init];
	}
	if ([name caseInsensitiveCompare:@"AdvGameBoard"]==0)
	{
		object=[[CRunAdvGameBoard alloc] init];
	}
	if ([name caseInsensitiveCompare:@"AdvPathMov"]==0)
	{
		object=[[CRunAdvPathMov alloc] init];
	}
	if ([name caseInsensitiveCompare:@"CalcRect"]==0)
	{
		object=[[CRunCalcRect alloc] init];
	}
	if ([name caseInsensitiveCompare:@"IIF"]==0)
	{
		object=[[CRunIIF alloc] init];
	}
	if ([name caseInsensitiveCompare:@"InAndOutController"]==0)
	{
		object=[[CRunInAndOutController alloc] init];
	}
	if ([name caseInsensitiveCompare:@"kcdirect"]==0)
	{
		object=[[CRunkcdirect alloc] init];
	}
	if ([name caseInsensitiveCompare:@"Layer"]==0)
	{
		object=[[CRunLayer alloc] init];
	}
	if ([name caseInsensitiveCompare:@"MoveSafely2"]==0)
	{
		object=[[CRunMoveSafely2 alloc] init];
	}
	if ([name caseInsensitiveCompare:@"Platform"]==0)
	{
		object=[[CRunPlatform alloc] init];
	}
	if ([name caseInsensitiveCompare:@"clickteam-movement-controller"]==0)
	{
		object=[[CRunclickteam_movement_controller alloc] init];
	}
	if ([name caseInsensitiveCompare:@"KcBoxA"]==0)
	{
		object=[[CRunKcBoxA alloc] init];
	}
	if ([name caseInsensitiveCompare:@"KcBoxB"]==0)
	{
		object=[[CRunKcBoxB alloc] init];
	}
	if ([name caseInsensitiveCompare:@"Accelerometer"]==0)
	{
		object=[[CRunAccelerometer alloc] init];
	}
	if ([name caseInsensitiveCompare:@"Location"]==0)
	{
		object=[[CRunLocation alloc] init];
	}
	if ([name caseInsensitiveCompare:@"kcini"]==0)
	{
		object=[[CRunkcini alloc] init];
	}
	if ([name caseInsensitiveCompare:@"MultipleTouch"]==0)
	{
		object=[[CRunMultipleTouch alloc] init];
	}
	if ([name caseInsensitiveCompare:@"iOSButton"]==0)
	{
		object=[[CRuniOSButton alloc] init];
	}
	if ([name caseInsensitiveCompare:@"iOSSingleEdit"]==0)
	{
		object=[[CRuniOSSingleEdit alloc] init];
	}
	if ([name caseInsensitiveCompare:@"iOSMultipleEdit"]==0)
	{
		object=[[CRuniOSMultipleEdit alloc] init];
	}
	if ([name caseInsensitiveCompare:@"AdvDir"]==0)
	{
		object=[[CRunAdvDir alloc] init];
	}
	if ([name caseInsensitiveCompare:@"JoystickControl"]==0)
	{
		object=[[CRunJoystickControl alloc] init];
	}
	if ([name caseInsensitiveCompare:@"kcclock"]==0)
	{
		object=[[CRunkcclock alloc] init];
	}
	if ([name caseInsensitiveCompare:@"kchisc"]==0)
	{
		object=[[CRunkchisc alloc] init];
	}
	if ([name caseInsensitiveCompare:@"kcrandom"]==0)
	{
		object=[[CRunkcrandom alloc] init];
	}
	if ([name caseInsensitiveCompare:@"ObjectMover"]==0)
	{
		object=[[CRunObjectMover alloc] init];
	}
	if ([name caseInsensitiveCompare:@"WargameMap"]==0)
	{
		object=[[CRunWargameMap alloc] init];
	}
	if ([name caseInsensitiveCompare:@"parse"]==0)
	{
		object=[[CRunparser alloc] init];
	}
	if ([name caseInsensitiveCompare:@"MTRandom"]==0)
	{
		object=[[CRunMTRandom alloc] init];
	}
	if ([name caseInsensitiveCompare:@"parser"]==0)
	{
		object=[[CRunparser alloc] init];
	}
	if ([name caseInsensitiveCompare:@"get"]==0)
	{
		object=[[CRunGet alloc] init];
	}
	if ([name caseInsensitiveCompare:@"kclist"]==0)
	{
		object=[[CRunkclist alloc] init];
	}
	if ([name caseInsensitiveCompare:@"kcinput"]==0)
	{
		object=[[CRunkcinput alloc] init];
	}
	if ([name caseInsensitiveCompare:@"iOSStore"]==0)
	{
		object=[[CRuniOSStore alloc] init];
	}	
	if ([name caseInsensitiveCompare:@"StringTokenizer"]==0)
	{
		object=[[CRunStringTokenizer alloc] init];
	}
	if ([name caseInsensitiveCompare:@"MoveIt"]==0)
	{
		object=[[CRunMoveIt alloc] init];
	}
	if ([name caseInsensitiveCompare:@"Easing"]==0)
	{
		object=[[CRunEasing alloc] init];
	}
	if ([name caseInsensitiveCompare:@"kcedit"]==0)
	{
		object=[[CRunkcedit alloc] init];
	}
	if ([name caseInsensitiveCompare:@"iOS"]==0)
	{
		object=[[CRuniOS alloc] init];
	}
	if ([name caseInsensitiveCompare:@"kcbutton"]==0)
	{
		object=[[CRunkcbutton alloc] init];
	}
	if ([name caseInsensitiveCompare:@"Camera"]==0)
	{
		object=[[CRunCamera alloc] init];
	}
	if ([name caseInsensitiveCompare:@"GameCenterConnect"]==0)
	{
		object=[[CRunGameCenterConnect alloc] init];
	}
	if ([name caseInsensitiveCompare:@"GameCenterLeaderboard"]==0)
	{
		object=[[CRunGameCenterLeaderboard alloc] init];
	}
	if ([name caseInsensitiveCompare:@"GameCenterAchievements"]==0)
	{
		object=[[CRunGameCenterAchievements alloc] init];
	}
	if ([name caseInsensitiveCompare:@"GameCenterAchievement"]==0)
	{
		object=[[CRunGameCenterAchievement alloc] init];
	}
	if ([name caseInsensitiveCompare:@"GameCenterMultiplayer"]==0)
	{
		object=[[CRunGameCenterMultiplayer alloc] init];
	}
	if ([name caseInsensitiveCompare:@"kcpica"]==0)
	{
		object=[[CRunkcpica alloc] init];
	}
	if ([name caseInsensitiveCompare:@"kcwctrl"]==0)
	{
		object=[[CRunkcwctrl alloc] init];
	}
	if ([name caseInsensitiveCompare:@"iOSVideo"]==0)
	{
		object=[[CRuniOSVideo alloc] init];
	}
	if ([name caseInsensitiveCompare:@"ForEach"]==0)
	{
		object=[[CRunForEach alloc] init];
	}
	if ([name caseInsensitiveCompare:@"CreateByName"]==0)
	{
		object=[[CRunCreateByName alloc] init];
	}
	if ([name caseInsensitiveCompare:@"Inventory"]==0)
	{
		object=[[CRunInventory alloc] init];
	}
	if ([name caseInsensitiveCompare:@"DeadReckoning"]==0)
	{
		object=[[CRunDeadReckoning alloc] init];
	}
	if ([name caseInsensitiveCompare:@"Flocking"]==0)
	{
		object=[[CRunFlocking alloc] init];
	}
	if ([name caseInsensitiveCompare:@"ScreenZoom"]==0)
	{
		object=[[CRunScreenZoom alloc] init];
	}
	if ([name caseInsensitiveCompare:@"Box2DBase"]==0)
	{
		object=[[CRunBox2DBase alloc] init];
	}
	if ([name caseInsensitiveCompare:@"Box2DFan"]==0)
	{
		object=[[CRunBox2DFan alloc] init];
	}
	if ([name caseInsensitiveCompare:@"Box2DMagnet"]==0)
	{
		object=[[CRunBox2DMagnet alloc] init];
	}
	if ([name caseInsensitiveCompare:@"Box2DTreadmill"]==0)
	{
		object=[[CRunBox2DTreadmill alloc] init];
	}
	if ([name caseInsensitiveCompare:@"Box2DParticules"]==0)
	{
		object=[[CRunBox2DParticules alloc] init];
	}
	if ([name caseInsensitiveCompare:@"Box2DRopeAndChain"]==0)
	{
		object=[[CRunBox2DRopeAndChain alloc] init];
	}
	if ([name caseInsensitiveCompare:@"Box2DGround"]==0)
	{
		object=[[CRunBox2DGround alloc] init];
	}
	if ([name caseInsensitiveCompare:@"Box2DJoint"]==0)
	{
		object=[[CRunBox2DJoint alloc] init];
	}
	if ([name caseInsensitiveCompare:@"WebView2"]==0)
	{
		object=[[CRunWebView2 alloc] init];
	}
	if ([name caseInsensitiveCompare:@"StencilMask"]==0)
	{
		object=[[CRunStencilMask alloc] init];
	}

	//F02END
	
	return object;
}

@end
